core/src/main/java/org/opensearch/sql/data/model/ExprTimeValue.java

```diff
@@ -51,6 +56,21 @@ public LocalTime timeValue() {
     return time;
   }
 
+  @Override
+  public LocalDate dateValue() {
+    return LocalDate.now();
```

Comment:
The assumed "current date" value must be the same for all temporal values in a query. Otherwise, a statement like `SELECT cast(TIME(13, 0) AS DATE) = cast(TIME(13, 0) AS DATE)` can have different result depending on when it is executed.

Reply:
@Yury-Fridlyand and @acarbonetto sorry I strongly disagree with this. This is buggy code and should not go into upstream like this.

At scale of a cloud service, someone will run into it and have no idea what happened.

Reply:
Fixed in 5468caa4b.